Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -221,10 +226,12 @@ private extension HTTPClient {
 
         return Result
             .success(dataIfAvailable(statusCode))
-            .mapSuccessToOptionalHTTPResult(statusCode)
+            .mapToResponse(response: httpURLResponse,
+                           request: request.httpRequest,
+                           signing: self.signing,
+                           verificationLevel: self.systemInfo.responseVerificationLevel)
             .map {
-                self.eTagManager.httpResultFromCacheOrBackend(with: httpURLResponse,
-                                                              data: $0.body,
+                self.eTagManager.httpResultFromCacheOrBackend(with: $0,
                                                               request: urlRequest,
                                                               retried: request.retried)
```

Comment:
This is based on the empty body case in https://github.com/RevenueCat/purchases-ios/pull/2272/files#diff-2a4a83f4e5d43494fe43c0b623281c54d24bd60193c7de185f37e565b0c72de3R107-R108 and the comment suggesting that code assumes a cache hit in the caller..

Reply:
> If there's an empty response and no etag, looks like this will be nil due to nothing in cache. Is that ok for callers?

Yeah that's confusing, it took me a while to wrap my head around why that was done when I refactored this months ago.
The answer to that is in `HTTPClient.handle(urlResponse:request:urlRequest:data:error:)`. If this returns `nil` (the scenario you described), it gets retried. There's plenty of test coverage for that.

> Is that ok for callers?

As for users of `HTTPClient`, ultimately responses can't be `nil` thanks to the strong type safety provided by its API.
The `Completion` type is defined as
```swift
Completion<Value: HTTPResponseBody> = (HTTPResponse<Value>.Result) -> Void
```

Which is basically `(Result<HTTPResponse<Value>, NetworkError>) -> Void`. In other words, there will be a response or an error, but never an empty or `nil` response.